src/eids.js

```diff
@@ -56,18 +56,16 @@ export class EidRewriter {
     this.idExemptButPassToChildren = [
       'arguments', 'background', 'conclusions', 'decision', 'header', 'intro', 'introduction', 'motivation', 'preamble', 'preface', 'remedies', 'wrapUp',
     ];
+    this.counters = {};
+    this.eIdCounter = {};
+    this.eIdMappings = {};
```

Comment:
don't we run the risk of having stale counters if we set those here? 
(and if not, why make a fresh mappings dict at 67?)

Reply:
A few aspects:

1. we're mimicking what we have in bluebell (which may not be ideal, but they must change together).
2. setting internal state up front is a good practice (rather than at the last minute); PyCharm sometimes warns that you're initialising a member variable outside of the constructor
3. we're not very consistent with our approach here: eidMappings is reset between calls, but these others aren't.

In terms of (3), we should either embrace that this is stateful and state is retained between calls (and so eIdMappings can be checked between calls, and the state must be reset between calls; my preference), or it should be stateless and every call is guaranteed to be fresh (and so state must be returned and then cleared). A mixed approach is confusing.

We'd have to check how both this and bluebell are called to before taking either path, which I don't really want to have to do right now :)